types/better-sqlite3/index.d.ts

```diff
@@ -13,6 +13,10 @@ type VariableArgFunction = (...params: any[]) => any;
 type ArgumentTypes<F extends VariableArgFunction> = F extends (...args: infer A) => any ? A : never;
 
 declare namespace BetterSqlite3 {
+    interface NamespacedDict {
+        [key: string]: object;
```

Comment:
fwiw, this is more specifically:
```suggestion
        [key: string]: string | number | Buffer | BigInt | null;
```

but I suspect changing this would be a bigger change than just this line

Reply:
Yeah, that's what I originally did, but thought that it'll require more maintaining with not so much of a payoff. If we are to make this change then bind parameters should also be changed to (string | number | Buffer | BigInt | null)[]

Reply:
👍 I _think_ object is equally good, it doesnt pollute the same way `any` does. E.g. if I have an untyped statement, I believe I need to cast a column as a number before I can use it as a number

Reply:
Should I make relevant changes or maybe postpone it for a different PR?

Reply:
Oh, actually, the changes would need to be made like
```javascript
interface Dict {
    [key: string]: string | number | Buffer | BigInt | null;
}
interface NamespacedDict {
    [key: string]: Dict
}```
Since **NamespacedDict** is used as return type in expanded mode. And then the object return value would need to bbe changed to **Dict** everywhere, including **BindParameters**

Reply:
> 👍 I _think_ object is equally good, it doesnt pollute the same way `any` does. E.g. if I have an untyped statement, I believe I need to cast a column as a number before I can use it as a number

Then I think if there's no immediate need in changing this it can be postponed :)

Reply:
yeah lets postpone

Reply:
Nevermind, seems like we'll have to do it properly. I'm not super experienced with typescript which is why I'm finding out about this on the go, lol. `object` is not an indexable type (doesn't have an index signature defined) which is why if you don't specify the return type you will have to cast the return to `any` first before indexing. I'll make the required changes and will push them soon.